### PR TITLE
Make Energy Evolution (Eevee SUM) handle timing conflict with Welder

### PR DESCRIPTION
**Pull request by starg09**
Related ruling:

> Q. If I use Welder and attach a Fire Energy to an Eevee that has the "Energy Evolution" Ability, do I draw the cards for Welder first or search for a Flareon due to "Energy Evolution" first (or is it a choice)?
> 
> A. Since they are simultaneous actions, it is the player's choice of which order they can do this. (May 9, 2019 TPCi Rules Team)

A similar logic would need to be added to the other Energy Evolution cards (Definitely the XY-Furious Fists Eevee, also the Neo Discovery and Unseen Forces prints whenever Unlimited includes SM sets)

### CODE REVIEW

File: src/tcgwars/logic/impl/gen7/SunMoon.groovy [hunk 1]
```diff
@@ -2423,17 +2423,42 @@ public enum SunMoon implements LogicCardInfo {
...
                     def sel=self.owner.pbg.deck.select(min:0, "Energy Evolution ${ef.card.basicType}",
                       {it.cardTypes.is(EVOLUTION) && it.types.contains(ef.card.basicType) && it.predecessor==self.name}, self.owner)
                     if(sel){
                       evolve(self, sel.first(), OTHER)
                     }
                     shuffleDeck(null, self.owner.toTargetPlayer())
                   }
+                  def welderChoice = 0
+                  if (welderFlag) {
+                    def welderChoice = choose([1, 2, 3], ["Use before drawing 3 cards", "Use after drawing 3 cards", "Don't use Energy Evolution"], "A basic Energy was attached to $self via Welder. When would you like to use Energy Evolution?")
```
Comment (luongthomas) [inline src/tcgwars/logic/impl/gen7/SunMoon.groovy:2445]: Build breaks here because `welderChoice` is already defined

Comment (starg09) [reply src/tcgwars/logic/impl/gen7/SunMoon.groovy]: Fixed, thanks!

**Comment by luongthomas** [comment]
I'm also getting an error when I pick option 2 in `welderChoice`
```
groovy.lang.MissingMethodException: No signature of method: tcgwars.logic.effect.EffectType.call() is applicable for argument types: (tcgwars.logic.impl.gen7.SunMoon$_getImplementation_closure101$_closure630$_closure632$_closure634$_closure636$_closure638) values: [tcgwars.logic.impl.gen7.SunMoon$_getImplementation_closure101$_closure630$_closure632$_closure634$_closure636$_closure638@60ab936d]
Possible solutions: wait(), name(), any(), wait(long), any(groovy.lang.Closure), each(groovy.lang.Closure)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:58) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:49) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125) ~[groovy-all-2.4.8.jar:2.4.8]
	at tcgwars.logic.impl.gen7.SunMoon$_getImplementation_closure101$_closure630$_closure632$_closure634$_closure636.doCall(SunMoon.groovy:2450) ~[classes/:na]
	at tcgwars.logic.impl.gen7.SunMoon$_getImplementation_closure101$_closure630$_closure632$_closure634$_closure636.doCall(SunMoon.groovy) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_251]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_251]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_251]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_251]
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93) ~[groovy-all-2.4.8.jar:2.4.8]
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294) ~[groovy-all-2.4.8.jar:2.4.8]
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1027) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117) ~[groovy-all-2.4.8.jar:2.4.8]
	at tcgwars.logic.groovy.TcgBuilders.delayed(TcgBuilders.groovy:452) ~[classes/:na]
	at tcgwars.logic.groovy.TcgBuilders.delayed(TcgBuilders.groovy) ~[classes/:na]
	at tcgwars.logic.groovy.TcgBuilders$delayed$9.callStatic(Unknown Source) ~[na:na]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:56) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:194) ~[groovy-all-2.4.8.jar:2.4.8]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:206) ~[groovy-all-2.4.8.jar:2.4.8]
```

**Comment by starg09** [comment]
oh my at that error 👀

**Comment by luongthomas** [comment]
Screen Shot 2020-07-29 at 4 21 39 PM
Maybe this can help

**Comment by starg09** [comment]
Oh, actually it does. Missed a comma lol

**Comment by luongthomas** [comment]
Screen Shot 2020-07-29 at 4 21 39 PM

**Comment by luongthomas** [comment]
I can confirm it works well

**Pull request merged**